File: 2024_1/Desenvolvimento de Sistemas Móveis/Aula02/atv01.dart
// Escreva um programa em DART que contém uma lista de 5 números inteiros como entrada e calcule a média desses números.
//Certifique-se de lidar com casos em que a lista pode estar vazia.

void main() {
  List<int> numbers = [10, 10, 10, 10, 10];
  if (numbers.isEmpty) {
    print("A lista está vazia.");
    return;
  }
  
  double sum = 0;
  for (int number in numbers) {
    sum += number;
  }
  
  double average = sum / numbers.length;
  
  print("A média dos números é: $average");
}

//

// Crie um programa que verifique uma variável utilizando o Switch, traduza as seguintes cores em inglês para o portugues: blue, red, green, yellow

void main() {
  String corEmIngles = 'red'; // altere a cor aqui para testar outras cores

  String corEmPortugues;

  switch (corEmIngles) {
    case 'blue':
      corEmPortugues = 'azul';
      break;
    case 'red':
      corEmPortugues = 'vermelho';
      break;
    case 'green':
      corEmPortugues = 'verde';
      break;
    case 'yellow':
      corEmPortugues = 'amarelo';
      break;
    default:
      corEmPortugues = 'Cor desconhecida';
  }

  print("A cor em português é: $corEmPortugues");
}

File: 2024_1/Tecnologias Móveis/Aula03/atv01.kt
// 1 - Escreva uma função que recebe uma lista de números e retorna a média aritmética.

fun calcularMedia(numeros: List<Double>): Double{
    if(numeros.isEmpty()){
        return 0.0;
    }

    var soma = 0.0;
    for(numero in numeros){
        soma += numero;
    }
    return soma / numeros.size
}

fun main (){
    val numeros = listOf(10.0, 10.0, 10.0, 10.0, 10.0);
    val media = calcularMedia(numeros);
    println("A média dos números é: $media");
}